packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2661,16 +2662,19 @@ function flushPassiveEffectsImpl() {
       // One solution is in the host config, if the passive effects
       // have not yet been run, make a call to flush the passive effects
       // right after paint.
-      const endTime = now();
       currentPendingTransitionCallbacks = null;
 
-      scheduleCallback(IdleSchedulerPriority, () =>
-        processTransitionCallbacks(
-          prevPendingTransitionCallbacks,
-          endTime,
-          prevRootTransitionCallbacks,
-        ),
-      );
+      setNextPaintCallback(endTime => {
```

Comment:
move this schedule call synchronous commit phase

Reply:
If they've already fired, then we can fire the callbacks. But if they haven't, we stash the time somewhere and when they do get flushed use it